eureka/src/main/java/com/linecorp/armeria/server/eureka/EurekaUpdatingListener.java

```diff
@@ -240,6 +244,19 @@ private static InstanceInfo fillAndCreateNewInfo(InstanceInfo oldInfo, Server se
                                 oldInfo.getLeaseInfo(), oldInfo.getMetadata());
     }
 
+    /**
+     * Strip any port in {@linkplain Server#defaultHostname()}, so that it can be used in Eureka
+     * registration as a {@code hostName} or {@code vipAddress}.
+     */
+    private static String withoutPort(String hostname) {
+        final int hostnameColonIdx = hostname.lastIndexOf(':');
+        if (hostnameColonIdx < 0) {
+            return hostname;
+        }
+
+        return hostname.substring(0, hostnameColonIdx);
```

Comment:
I guess I owe a test here. need to make a server that has a VirtualHost that includes a non-zero port.. or.. expose this package private and test directly. which do you prefer?

Reply:
or if `Server.defaultHostame()` could never be derived from a VirtualHost with a non-zero port... let me know why and I'll comment the same!